diff --git a/libraries/lib-utility/MemoryX.h b/libraries/lib-utility/MemoryX.h
index da299a1bc..ddfe78e38 100644
--- a/libraries/lib-utility/MemoryX.h
+++ b/libraries/lib-utility/MemoryX.h
@@ -5,6 +5,7 @@
 #include <iterator>
 #include <memory>
 #include <cstdlib> // Needed for free.
+#include <limits>  // need for GCC 11.2.0 patch security features of GLIBC 2.33
 #ifndef safenew
 #define safenew new
 #endif
